tests/test_device.py

```diff
@@ -403,6 +424,35 @@ def circuit_gauss(mag_alpha, phase_alpha, phi):
             node_gauss(0.015, 0.02, 0.005)
         assert dev_gauss.num_executions == num_evals_gauss
 
+    @pytest.mark.parametrize(
+        "depth, expanded_ops",
+        [
+            (0, [qml.PauliX(0), qml.BasisEmbedding([1, 0], wires=[1, 2])]),
+            (1, [qml.PauliX(wires=0), qml.PauliX(wires=1)]),
+        ],
+    )
+    def test_device_default_expand_ops(
+        self, depth, expanded_ops, mock_device_with_paulis_hamiltonian_and_methods
+    ):
+        """Test that the default expand method can selectively expand operations
+        without expanding measurements."""
```

Comment:
Does this test captures the whole functionality of the newly added method?

Reply:
Yes it should 👍🏼